controller/service.go

```diff
@@ -30,14 +30,40 @@ import (
 )
 
 const (
-	AnnotationPrefix             = "metallb.universe.tf"
+	AnnotationPrefix             = "metallb.io"
 	AnnotationAddressPool        = AnnotationPrefix + "/" + "address-pool"
 	AnnotationLoadBalancerIPs    = AnnotationPrefix + "/" + "loadBalancerIPs"
 	AnnotationIPAllocateFromPool = AnnotationPrefix + "/" + "ip-allocated-from-pool"
+	AnnotationAllowSharedIP      = AnnotationPrefix + "/" + "allow-shared-ip"
+
+	// Deprecated Annotations. Used for backward compatibility.
+	DeprecatedAnnotationPrefix             = "metallb.universe.tf"
+	DeprecatedAnnotationAddressPool        = DeprecatedAnnotationPrefix + "/" + "address-pool"
+	DeprecatedAnnotationLoadBalancerIPs    = DeprecatedAnnotationPrefix + "/" + "loadBalancerIPs"
+	DeprecatedAnnotationIPAllocateFromPool = DeprecatedAnnotationPrefix + "/" + "ip-allocated-from-pool"
+	DeprecatedAnnotationAllowSharedIP      = DeprecatedAnnotationPrefix + "/" + "allow-shared-ip"
 )
 
 var ErrConverge = fmt.Errorf("failed to converge")
 
+// SharingKey extracts the sharing key for a service.
+func SharingKey(svc *v1.Service) string {
```

Comment:
why not changing the implementation in k8salloc?

Reply:
My modification uses some of the constants defined in service.go (package controller). However, importing controller into k8salloc produces an import loop and I don't liked the idea of defining the annotations multiple times in different packages. Moreover, SharingKey(...) is only used in service.go.